packages/ui/Slottable/Slottable.js

```diff
@@ -103,16 +43,38 @@ const defaultConfig = {
  */
 const Slottable = hoc(defaultConfig, (config, Wrapped) => {
 	const slots = config.slots;
-	return kind({
-		name: 'Slottable',
-		render: (props) => {
-			const adjusted = distribute(slots, props);
-			return (
-				<Wrapped {...adjusted} />
-			);
-		}
-	});
+
+	// eslint-disable-next-line no-shadow
+	return function Slottable (props) {
+		// extract the slots into a new object but populating the default value to be null to allow
+		// the current "harmful" behavior below.
+		const slotProps = {children: props.children};
+		slots.forEach(k => (slotProps[k] = null));
+
+		// Slottable allows there to be other values in the destination slot and merges them.
+		// However, consumers can't avoid key warnings when merging the two lists so we should
+		// "consider this harmful" and not continue to support this with the hook and instead
+		// encourage using the slot as the default with the prop as a fallback as implemented by the
+		// hook.
+		const distributed = useSlots(slotProps);
+		slots.forEach(slot => {
```

Comment:
Do we need to ensure that `children` never appears in `slots`?

Reply:
We don't currently but that's a reasonable condition to add.

Reply:
Actually, I'd like to skip this. We don't test for this currently and it isn't possible with the hook implementation so it's out of scope and really edge case-y.

Reply:
Makes sense!  Was just asking as food for thought.